File: CSharp/DependencyBreakingTests/ParameteriseConstructor/DiscountTest.cs
using Org.Codecop.Dependencies.ParameteriseConstructor;
using Xunit;

namespace Org.Codecop.Dependencies.ParameteriseConstructor.Tests
{
    public class DiscountTest
    {
        [Fact]
        public void Test1()
        {
            var discount = new Discount();

            var net = new Money(1002);
            var total = discount.DiscountFor(net);

            Assert.Equal(new Money(901.8m), total);
        }
    }
}

File: Java/src/test/java/org/codecop/dependencies/a/DiscountTest.java
package org.codecop.dependencies.a;

import java.math.BigDecimal;
import org.junit.Test;

import static org.junit.Assert.assertEquals;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

public class DiscountTest {

    @Test
    public void discountForMoreThanThousand() {
        MarketingCampaign marketingCampaign = mock(MarketingCampaign.class);
        when(marketingCampaign.isCrazySalesDay()).thenReturn(false);
        Discount discount = new Discount(marketingCampaign);

        Money net = new Money(1002);
        Money total = discount.discountFor(net);

        assertEquals(new Money(new BigDecimal("901.8")), total);
    }

    @Test
    public void discountForCrazySalesDay() {
        MarketingCampaign marketingCampaign = mock(MarketingCampaign.class);
        when(marketingCampaign.isCrazySalesDay()).thenReturn(true);
        Discount discount = new Discount(marketingCampaign);
        
        Money net = new Money(100);
        Money total = discount.discountFor(net);
        
        assertEquals(new Money(85), total);
    }

}
